src/util/ConfigStore.cc

```diff
@@ -113,25 +112,44 @@ int ConfigStore::getIntRequired(std::string const& key) const {
     throw util::KeyNotFoundError(key);
 }
 
+std::set<std::string> ConfigStore::getSections() const {
+    std::set<std::string> sections;
+    for (auto [key, val] : _configMap) {
+        if (auto const pos = key.find("."); pos != std::string::npos) {
+            sections.insert(key.substr(0, pos));
+        }
+    }
+    return sections;
+}
+
 std::map<std::string, std::string> ConfigStore::getSectionConfigMap(std::string sectionName) const {
-    // find all css.* parameters and copy to new map (dropping css.)
-    std::string section = sectionName + ".";
+    // Copy all parameters matching "<sectionName>.<param>" to new map (dropping "<sectionName>.")
+    std::string const section = sectionName + ".";
     std::map<std::string, std::string> sectionConfigMap;
     int len = section.length();
-    for (auto const& kv : _configMap) {
-        if (kv.first.compare(0, len, section) == 0) {
-            sectionConfigMap.insert(std::make_pair(std::string(kv.first, 4), kv.second));
+    for (auto&& [key, val] : _configMap) {
+        if (key.substr(0, len) == section) {
+            sectionConfigMap[key.substr(len)] = val;
```

Comment:
I like the change, but I think `auto const&` would be better than `auto&&` in this case, as `auto&&` could allow `val` to be altered.

Reply:
Well, `auto&&` in this context means "forwarding" of whatever types are returned by the range iterator. This is just the "lazy" way of writing `auto const&`. Neither `key` nor `val` could be modified in this context because a context (a method) where this iteration is happening is `const`.

Reply:
No, auto&& preserves constantness. If it's a constant going in, it stays a constant. If it isn't a constant going in, then it isn't constant.
```
    std::map<string, int> numMap;
    numMap.insert(pair("one",1));
    numMap.insert(pair("two", 2));
    for (auto&& [key, val] : numMap) {
        val = 1234;
    }
```

Reply:
Your example is incorrect because the iterable collection isn't `const`. If you rewrite your code like this then it won't compile:
```
std::map<string, int> const numMap = {{"one",1}, {"two", 2}};
for (auto&& [key, val] : numMap) {
    val = 1234;   // <- compilation ERROR
}
```
In the above-shown scenario, both `key` and `val` look like this:
```
std::string const& key = ...
int const& val = ...
```
Back to the original comment. The simplified version of my code in this PR looks like this:
```
ConfigStore {
public:
    std::map<std::string, std::string> getSectionConfigMap(std::string sectionName) const {
        for (auto&& [key, val] : _configMap) {
            ...
        }
    }
private:
    std::map<std::string, std::string> const _configMap;
};
```
With such a definition, `_configMap` (as well as `key` and `val`) become immutable for two reasons:
- because of the `const` attribute of the method `getSectionConfigMap`
- and because the data member by itself is `const`
- Therefore type forwarding made by `auto&&` does the right thing (forwards both to be `std::string const&`).